testsuite/MDAnalysisTests/analysis/test_encore.py

```diff
@@ -826,20 +827,22 @@ def test_get_distance_matrix(self):
         dm = confdistmatrix.get_distance_matrix(u)
 
 class TestEncoreImportWarnings(object):
-    def setUp(self):
-        # clear cache of encore module
+    @block_import('sklearn')
+    def _check_sklearn_import_warns(self, package, recwarn):
         for mod in list(sys.modules):  # list as we're changing as we iterate
             if 'encore' in mod:
                 sys.modules.pop(mod, None)
-
-    @block_import('sklearn')
-    def _check_sklearn_import_warns(self, package):
         warnings.simplefilter('always')
-        assert_warns(ImportWarning, importlib.import_module, package)
+        # assert_warns(ImportWarning, importlib.import_module, package)
+        importlib.import_module(package)
+        assert recwarn.pop(ImportWarning)
 
-    def test_import_warnings(self):
+    def test_import_warnings(self, recwarn):
+        for mod in list(sys.modules):  # list as we're changing as we iterate
+            if 'encore' in mod:
+                sys.modules.pop(mod, None)
         for pkg in (
                 'MDAnalysis.analysis.encore.dimensionality_reduction.DimensionalityReductionMethod',
                 'MDAnalysis.analysis.encore.clustering.ClusteringMethod',
         ):
-            yield self._check_sklearn_import_warns, pkg
+            self._check_sklearn_import_warns(pkg, recwarn)
```

Comment:
This changes the behaviour of the test. Instead of having 2 tests that can fail individually, there is only one test that has two consecutive asserts. It is fine for now, but please add a comment as it may be forgotten when you will move to proper pytest idioms.